cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractListener.java

```diff
@@ -27,7 +27,9 @@
 
   @Override
   public void init(ProgramController.State currentState, @Nullable Throwable cause) {
-    if (currentState == ProgramController.State.COMPLETED) {
+    if (currentState == ProgramController.State.ALIVE) {
```

Comment:
This makes sure that the state is not missed on the controller - I believe it is still needed, as tests were failing without it.